src/include/distributed/causal_clock.h

```diff
@@ -0,0 +1,44 @@
+/*
+ * causal_clock.h
+ *
+ * Data structure definitions for managing hybrid logical clock and
+ * related function declarations.
+ *
+ * Copyright (c) Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#ifndef CAUSAL_CLOCK_H
+#define CAUSAL_CLOCK_H
+
+#include "distributed/type_utils.h"
+
+/*
+ * Clock components - Unsigned 64 bit <LC, C>
+ * Logical clock   (LC): 42 bits
+ * Counter          (C): 22 bits
+ *
+ * 2^42 milliseconds - 4398046511104 milliseconds, which is ~139 years.
+ * 2^22 ticks - maximum of four million operations per millisecond.
+ *
+ */
+
+#define LOGICAL_BITS 42
+#define COUNTER_BITS 22
+#define LOGICAL_MASK ((1U << COUNTER_BITS) - 1)
+
+#define MAX_LOGICAL ((1LU << LOGICAL_BITS) - 1)
```

Comment:
Hmm actually this one is still used, so maybe we do enforce the 42 bit space currently.
Is there a technical reason to keep the logical clock limited to 2^42?

Reply:
There is no technical reason to limit the logical clock to 2^42, but it's good to have some limit to find any runaway clocks. Maybe I misunderstood your earlier comment on the same topic, here you meant 2^64 for logical and not combined?


![image](https://user-images.githubusercontent.com/44680808/198691626-dafe811b-1e22-4437-bfa5-9e20c867bedd.png)

![image](https://user-images.githubusercontent.com/44680808/198691732-86a068f6-7561-4fde-9db9-1bca084dde17.png)